packages/next/src/server/app-render/app-render.tsx

```diff
@@ -5121,3 +5160,21 @@ async function collectSegmentData(
     serverConsumerManifest
   )
 }
+
+function isBypassingCachesInDev(
+  renderOpts: RenderOpts,
+  requestStore: RequestStore
+) {
+  return (
+    process.env.NODE_ENV === 'development' &&
+    renderOpts.dev &&
+    requestStore.headers.get('cache-control') === 'no-cache'
+  )
+}
```

Comment:
nit: let's make it return `boolean` not `boolean | undefined` (apparently `renderOpts.dev` is optional? weird)
```suggestion
function isBypassingCachesInDev(
  renderOpts: RenderOpts,
  requestStore: RequestStore
): boolean {
  return (
    process.env.NODE_ENV === 'development' &&
    !!renderOpts.dev &&
    requestStore.headers.get('cache-control') === 'no-cache'
  )
}
```

Reply:
also, tbh i'd get rid of `renderOpts.dev` here altogether, we're already branching on `process.env.NODE_ENV` basically everywhere, it doesn't add anything afaict

Reply:
No, there are "log extra info in dev" forks and then there are "do specific behavior when running `next dev`".

Really we'll never support next dev with NODE_ENV !== "development"
but we could support next start with NODE_NEV === "development".

I really think we need to encode this statically vs as a renderOpt though and then build against it. So like have

`__DEV_SERVER__` as a global that we can statically replace in both our server-runtime bundle and in the user code bundle (only because some Next.js code is only loaded here)